lib/xgu/xgux.h

```diff
@@ -0,0 +1,169 @@
+#ifndef __XGUX_H
+#define __XGUX_H
+
+#include "xgu.h"
+
+#define MIN(a,b) ((a)<(b)?(a):(b))
+#define MAX_BATCH 120
```

Comment:
This needs a comment. I believe I wanted this to multiple of 2,3,4,5 (so all [?] primitive types work). If this is done, it comes down to: 2*3*4*5 which happens to be 120.
We could potentially have a different batch count per primitive type to make this more optimal (larger batches = less CPU time and probably less GPU setup overhead).